src/Manager.php

```diff
@@ -52,80 +74,105 @@ public function table()
+    public function collection($name = null)
     {
-        $this->_config[$name] = new Type\Value($name, $this, $config);
+        if ($name === null) {
+            return $this->_collection;
+        }
+        if (!isset($this->_filters[$name])) {
+            $this->_filters[$name] = [];
+        }
+        $this->_collection = $name;
         return $this;
     }
 
     /**
-     * finder method
+     * Adds a new filter to the active collection.
      *
-     * @param string $name Name
-     * @param array $config Config
+     * @param string $name Field name.
+     * @param string $filter Filter name.
+     * @param array $options Filter options.
      * @return $this
      */
-    public function finder($name, array $config = [])
+    public function add($name, $filter, array $options = [])
     {
-        $this->_config[$name] = new Type\Finder($name, $this, $config);
+        $this->_filters[$this->_collection][$name] = $this->loadFilter($name, $filter, $options);
         return $this;
     }
 
     /**
-     * callback method
+     * Removes filter from the active collection.
      *
-     * @param string $name Name
-     * @param array $config Config
-     * @return $this
+     * @param string $name Name of the filter to be removed.
+     * @return void
      */
-    public function callback($name, array $config = [])
+    public function remove($name)
     {
-        $this->_config[$name] = new Type\Callback($name, $this, $config);
-        return $this;
+        unset($this->_filters[$this->_collection][$name]);
     }
 
     /**
-     * compare method
+     * Loads a search filter.
      *
-     * @param string $name Name
-     * @param array $config Config
-     * @return $this
+     * @param string $name Name of the field
+     * @param string $filter Filter name
+     * @param array $options Filter options.
+     * @return \Search\Type\Base
+     * @throws \InvalidArgumentException When no filter was found.
      */
-    public function compare($name, array $config = [])
+    public function loadFilter($name, $filter, array $options = [])
     {
-        $this->_config[$name] = new Type\Compare($name, $this, $config);
-        return $this;
+        list($plugin, $filter) = pluginSplit($filter);
+        $filter = Inflector::classify($filter);
+        if (!empty($plugin)) {
+            $className = '\\' . $plugin . '\Search\Type\\' . $filter;
+            if (class_exists($className)) {
+                return new $className($name, $this, $options);
+            }
+        }
+        $typeClasses = $this->config('typeClasses');
+        if (isset($typeClasses[$filter])) {
+            return new $typeClasses[$filter]($filter, $this, $options);
+        }
+        if (class_exists('\Search\Type\\' . $filter)) {
+            $className = '\Search\Type\\' . $filter;
+            return new $className($name, $this, $options);
+        }
+        if (class_exists('\App\Search\Type\\' . $filter)) {
```

Comment:
Looks like `App::className()` requires/expects you to use the plugin name, otherwise it will always search within the App namespace. `App::className($filter, 'Search/Type');` does not work, but `App::className('Search.' . $filter, 'Type');` does.

```php
$className = App::className($filter, 'Search/Type');
if (!$className) {
    $className = App::className('Search.' . str_replace('Search.', '', $filter), 'Type');
}
```

Should work as expected.